pyccl/tests/test_ept_power.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import pyccl as ccl
+import pytest
+
+
+COSMO = ccl.Cosmology(
+    Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+    transfer_function='bbks', matter_power_spectrum='linear')
+TRS = {'TG': ccl.nl_pt.PTNumberCountsTracer(b1=2.0, b2=2.0, bs=2.0),
+       'TI': ccl.nl_pt.PTIntrinsicAlignmentTracer(c1=2.0, c2=2.0,
+                                                  cdelta=2.0),
+       'TM': ccl.nl_pt.PTMatterTracer()}
+PTC = ccl.nl_pt.EulerianPTCalculator(with_NC=True, with_IA=True,
+                                     with_matter_1loop=True,
+                                     cosmo=COSMO)
+
+
+def test_ept_calculator_smoke():
+    c = ccl.nl_pt.EulerianPTCalculator(log10k_min=-3,
+                                       log10k_max=1,
+                                       nk_per_decade=10,
+                                       extra_params={'pad_factor': 2})
+    assert len(c.k_s) == 40
+
+
+@pytest.mark.parametrize('options', [['TG', 'TG', False, False],
+                                     ['TG', 'TG', False, True],
+                                     ['TG', 'TI', False, False],
+                                     ['TG', 'TM', False, False],
+                                     ['TI', 'TG', False, False],
+                                     ['TI', 'TI', False, False],
+                                     ['TI', 'TI', True, False],
+                                     ['TI', 'TM', False, False],
+                                     ['TM', 'TG', False, False],
+                                     ['TM', 'TI', False, False],
+                                     ['TM', 'TM', False, False]])
+def test_ept_get_pk2d_smoke(options):
+    if options[0] == options[1]:
+        t2 = None
+    else:
+        t2 = TRS[options[1]]
+    ptc = ccl.nl_pt.EulerianPTCalculator(
+        with_NC=True, with_IA=True, with_matter_1loop=True,
+        extra_params={'sub_lowk': options[3]},
+        cosmo=COSMO)
+    pk = ptc.get_pk2d_biased(TRS[options[0]],
+                             tracer2=t2,
+                             return_ia_bb=options[2])
+    assert isinstance(pk, ccl.Pk2D)
```

Comment:
I have already replaced some of the tests that do this, but this way of writing tests is very hard to read, and if one fails in the future the developer will have trouble debugging it. It is best to unpack `options` into the real parameter names, and use it like so:
```python
@pytest.mark.parametrize("param1,param2,param3,param4",  # replace with real names
    [["TG", "TG", ...],
    ...]])
def test_(param1, param2, param3, param4):
    ...
```

Reply:
Done